connection/src/thick.rs

```diff
@@ -156,6 +159,7 @@ pub struct ThickClient<CP: CredentialsProvider> {
 impl<CP: CredentialsProvider> ThickClient<CP> {
     /// Create a new attested connection to the given consensus node.
     pub fn new(
+        network_id: String,
```

Comment:
I wonder if this should not be a query parameter in the uri. We already use it for other things like responder id, tls config, etc

Reply:
that seems fine to me

do you think `MC_NETWORK_ID` should also go away for the clients? or clients should add that to the uri if there is no query parameter in the uri

do you think `MC_NETWORK_ID` should go away for the servers?

Reply:
for some things like `GrpcFogReportConnection` i think the only way that makes sense is the client software sets the network id, and not the uri from your counterparty. otherwise your software will happily mix networks if you paste someone's address from another network

Reply:
i guess the thing i'm thinking about now is, most clients have several URIs (multiple consensus nodes, or, a bunch of fog servers, etc.) its kind of nice if there is a single source of truth for network id. we could do something like, extract network id from all of the uris, and say it's an error if any of them don't match?

Reply:
> We can add a convenience .with_network_id() method that returns a copy of the URI with the network-id parameter added for cases where you only want to pass it once (e.g. mobilecoind peers). 

Saw this now, thinking about it

Reply:
I guess having thought about it now, I feel like it's better not to involve the uris and just do it similar to how I did it so far.

* We want clients and servers to have a single source of truth about the network id, so MC_NETWORK_ID as a string seems like the simplest way to do that
* Ideally, as a programmer, I can trace from MC_NETWORK_ID, or some other source of truth, more or less directly to the header value, so that there is no confusion about why I'm getting a mismatch error
* It's not super interesting to have NETWORK_ID varying on a per-connection basis, I think that will always be a bug. But that is exactly what is allowed by making it a query parameter?

What are the pros of making it a uri query parameter?

* Fewer arguments to constructors like the one you are commenting on (but this seems minor)
* Clients and servers that are already getting uris on the command line don't necessarily have to change anything about their commnand line interface. But then, we have to extract the network ids and ensure that they all match to prevent misuse, which is probably more code and complexity overall.